metricbeat/helper/prometheus/prometheus.go

```diff
@@ -40,21 +40,33 @@ type Prometheus interface {
 	ReportProcessedMetrics(mapping *MetricsMapping, r mb.ReporterV2)
 }
 
+// EventLayout defines the structural option of how metrics are transformed into events
+type EventLayout int
+
+const (
+	// StandardLayout will group metrics with same name and keylabels into an event
+	StandardLayout EventLayout = iota
+	// ExpandedBucketsLayout will use expand Histograms and Summaries from the StandardLayout
+	// and generate an event for each histogram `le` value and summary `quantile`
+	ExpandedBucketsLayout
```

Comment:
I hope we can get opinionated here, using always the same layout for consistency. is this added for backwards compatibility? Do you expect new metricsets using the standard layout?

Reply:
yes, this is added for backwards compatibility, not being force to replace all current usages + re-create dashboards as this get merged